src/sources/dunamu.rs

```diff
@@ -0,0 +1,175 @@
+//! Dunamu Source Provider (v0.4 API)
+//! <https://www.dunamu.com/>
+
+use crate::{
+    error::{Error, ErrorKind},
+    prelude::*,
+};
+use crate::{Currency, Price, TradingPair};
+use bytes::buf::ext::BufExt;
+use hyper::{
+    client::{Client, HttpConnector},
+    header, Body, Request,
+};
+use hyper_rustls::HttpsConnector;
+use serde::{Deserialize, Serialize};
+
+//https://quotation-api-cdn.dunamu.com/v1/forex/recent?codes=FRX.KRWUSD
+
+/// Base URI for requests to the Coinone API
+pub const BASE_URI: &str = "https://quotation-api-cdn.dunamu.com";
+
+/// User-Agent to send in HTTP request
+pub const USER_AGENT: &str = "iqlusion delphi";
+
+/// Source provider for Dunamu
+pub struct DunamuSource {
+    http_client: Client<HttpsConnector<HttpConnector>>,
+}
+
+impl DunamuSource {
+    /// Create a new Dunamu source provider
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        Self {
+            http_client: Client::builder().build(HttpsConnector::new()),
+        }
+    }
+
+    /// Get trading pairs
+    pub async fn trading_pairs(&self, pair: &TradingPair) -> Result<Response, Error> {
+        if pair.0 != Currency::Krw && pair.1 != Currency::Krw {
+            fail!(ErrorKind::Currency, "trading pair must be with KRW");
+        }
+
+        let uri = format!(
+            "{}/v1/forex/recent?codes=FRX.{}{}",
+            BASE_URI, pair.0, pair.1
+        );
+
+        let mut request = Request::builder()
+            .method("GET")
+            .uri(&uri)
+            .body(Body::empty())?;
+
+        {
+            let headers = request.headers_mut();
+            headers.insert(header::CONTENT_TYPE, "application/json".parse().unwrap());
+            headers.insert(
+                header::USER_AGENT,
+                format!("{}/{}", USER_AGENT, env!("CARGO_PKG_VERSION"))
+                    .parse()
+                    .unwrap(),
+            );
+        }
+
+        let response = self.http_client.request(request).await?;
+        let body = hyper::body::aggregate(response.into_body()).await?;
+        Ok(serde_json::from_reader(body.reader())?)
+    }
+}
+
+/// API responses Vector
+pub type Response = Vec<ResponseElement>;
+/// API response entity
+#[derive(Serialize, Deserialize)]
+pub struct ResponseElement {
```

Comment:
For `camelCase` APIs like this, you can do the following which avoids the need to annotate each field separately:

```suggestion
#[serde(rename_all = "camelCase")]
pub struct ResponseElement {
```